[E2E/Acceptance CI Failure]: should draft, discard and publish the changes (exploration-editor/save-draft-publish-and-discard-the-channges)

### CI Test Type

Acceptance

### Which CI step failed?

should draft, discard and publish the changes 

### Did the test fail after an automatic rerun?

Test failed once and then passed after an automatic rerun

### Did the test fail when running on desktop, mobile, or both?

Mobile

### Stacktrace

```shell
FAIL core/tests/puppeteer-acceptance-tests/specs/exploration-editor/save-draft-publish-and-discard-the-changes.spec.ts (136.789 s)
  Exploration Creator
    ✕ should draft, discard and publish the changes (80993 ms)

  ● Exploration Creator › should draft, discard and publish the changes

    TimeoutError: waiting for selector `button.e2e-test-confirm-discard-changes` failed: timeout 30000ms exceeded

      3625 |       await this.clickOnElementWithSelector(desktopDiscardDraftButton);
      3626 |     }
    > 3627 |     await this.page.waitForSelector(confirmDiscardButton, {
           |                     ^
      3628 |       visible: true,
      3629 |     });
      3630 |     await Promise.all([

      at new WaitTask (../../../node_modules/puppeteer/src/common/DOMWorld.ts:813:28)
      at DOMWorld.waitForSelectorInPage (../../../node_modules/puppeteer/src/common/DOMWorld.ts:656:22)
      at Object.internalHandler.waitFor (../../../node_modules/puppeteer/src/common/QueryHandler.ts:78:19)
      at DOMWorld.waitForSelector (../../../node_modules/puppeteer/src/common/DOMWorld.ts:511:25)
      at Frame.waitForSelector (../../../node_modules/puppeteer/src/common/FrameManager.ts:1290:47)
      at Page.waitForSelector (../../../node_modules/puppeteer/src/common/Page.ts:3222:29)
      at BaseUser.discardCurrentChanges (utilities/user/exploration-editor.ts:3627:21)
          at runMicrotasks (<anonymous>)
      at Object.<anonymous> (specs/exploration-editor/save-draft-publish-and-discard-the-changes.spec.ts:72:7)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        138.404 s, estimated 146 s
Ran all test suites matching /core\/tests\/puppeteer-acceptance-tests\/specs\/exploration-editor\/save-draft-publish-and-discard-the-changes.spec.ts/i.
INFO     2025-12-17 21:38:39,857 shutdown.py:48] Shutting down.
Error: 2-17 21:38:39 +0000] [4537] [ERROR] Worker (pid:4545) was sent SIGTERM!
Error: 2-17 21:38:39 +0000] [4381] [ERROR] Worker (pid:4383) was sent SIGTERM!
INFO     2025-12-17 21:38:39,955 stub_util.py:321] Applying all pending transactions and saving the datastore
INFO     2025-12-17 21:38:39,956 stub_util.py:324] Saving search indexes
```

### Screenshots and/or Screen Recordings

https://github.com/user-attachments/assets/cc8c20e4-fbfc-4cdc-b8ef-5ff80bd7e7cc

### Occurrences

https://github.com/oppia/oppia/actions/runs/20316416714/job/58362367185?pr=24059

### Additional Information

_No response_

### Debugging document link

_No response_

--- a/core/tests/puppeteer-acceptance-tests/utilities/user/exploration-editor.ts
+++ b/core/tests/puppeteer-acceptance-tests/utilities/user/exploration-editor.ts
@@ -3613,6 +3613,9 @@ export class ExplorationEditor extends BaseUser {
         visible: true,
       });
       await this.clickOnElementWithSelector(mobileChangesDropdownSelector);
+      await this.page.waitForSelector(
+        `${mobileDiscardButtonSelector}:not(.disabled)`
+      );
       await this.clickOnElementWithSelector(mobileDiscardButtonSelector);
     } else {
       await this.page.waitForSelector(discardDraftDropdownSelector, {
@@ -3622,6 +3625,9 @@ export class ExplorationEditor extends BaseUser {
       await this.page.waitForSelector(desktopDiscardDraftButton, {
         visible: true,
       });
+      await this.page.waitForSelector(
+        `${desktopDiscardDraftButton}:not(.disabled)`
+      );
       await this.clickOnElementWithSelector(desktopDiscardDraftButton);
     }
     await this.page.waitForSelector(confirmDiscardButton, {

